### ISSUE TITLE
Where is...

### ISSUE THREAD

**Issue by dynarian** [iquidus/explorer#446]
I have translated the whole file "en.json" but there is a text that I have not been able to translate. Attached image marking these texts. Where can I modify them?

![imagen014](https://user-images.githubusercontent.com/37242241/110388944-21f94000-8064-11eb-846f-7aeff718530c.jpg)

**Comment by joeuhren** [comment]
Those values are generated from the [DataTables](https://datatables.net) library. Looks like they haven't been officially localized in iquidus yet, but since you're already making custom changes to the code, you can take a look here for the manual which shows some examples of how to override those values: https://datatables.net/manual/i18n

I would do a search through the `views` directory for `.dataTable(` to find all affected tables and you will need to add the `language` object to all of them and override all or whatever values you want by following the syntax in from the DataTables manual. Simple example:

```
    ......... .dataTable({
    language: {
      lengthMenu: "Mostrar _MENU_ entradas",
      info: "Mostrando _START_ a _END_ de _TOTAL_ entradas",
      paginate: {
            first: "Primero",
            previous: "Previo",
            next: "Próximo",
            last: "Último"
        }
    },
    .
    .
    .
    .
    });
```

Please note that I didn't actually test this out myself, but the instructions are pretty clear and I expect it shouldn't be difficult to make it work.

**Comment by dynarian** [comment]
joeuhren,thanksfor U comments,but I am not able to find .datatable in the explorer directory, what I do see is that in the view directory in the layout.jade file I have seen that it loads cdn.datatables.net/plug-ins/725b2a2115b/integration/bootstrap/3/ dataTables.bootstrap.js but it is something external to the explorer directories

**Comment by joeuhren** [comment]
Sorry for the confusion. What I meant is that you can do a search though the `.pug` files in the `views` directory to find all the places that DataTables are used. For example `views/movement.pug` has one instance of the word/phrase `.dataTable(` inside the code on line 47. You'll need to find all those same lines within all the `.pug` files and add the `language` object inside them, similar to how I demonstrated above, although you'll want to reference the DataTables manual as well. There's less than 10 DataTables, but I would recommend finding and starting with one until you can get it right and then you can apply the same fix to all of them.

Also, I noticed you mentioned the `layout.jade` file, so you must be using an older version of iquidus, as `jade` has been replaced by `pug` a while ago. I assume the same fix would apply to the older version, but just keep in mind you may want to look into upgrading at some point.

**Comment by dynarian** [comment]
Ok, thanks for your message. I cannot migrate to the .pug version since I have only one node of my coin working and it also depends on a POS development that gives the coin usability. I tried to do what you tell me in one of the movement.jade ps files, but I can't get it to work, I always get an error.

code example ...

      var rtable = $('#alerts-table').dataTable( { 
      paginate: {
            previous:   "Anterior",
            next:       "Siguiente"
        },
        autoWidth: false,      
        searching: false,
        ordering: false,
        //responsive: true,
        lengthChange: true,
        processing: true,
        serverSide: true,
        iDisplayLength: displayLengthMax,
        lengthMenu: lengthMenuOpts,
        //processing: true,
        ajax: '/ext/getlasttxsajax/#{min_amount}',
        rowCallback: function(row, data, index) {
          var blockindex = data[0]; //variables for better readability
          var blockhash = data[1]; //variables for better readability
          var txhash = data[2]; //variables for better readability
          var outputs = data[3]; //variables for better readability
          var amount = (data[4] / 100000000); //variables for better readability
          var amountParts = amount.toLocaleString('en',{'minimumFractionDigits':2,'maximumFractionDigits':8,'useGrouping':true}).split('.');
          var amountStr = amountParts[0] + '.<span class="decimal">' + amountParts[1] + '</span>';
          var timestamp = data[5]; //variables for better readability
          if (amount > '#{flagb}') {
            var total = "<label class='label label-danger'>" + amountStr + "</label>";
          } else if (amount > '#{flaga}') {
            var total = "<label class='label label-warning'>" + amountStr + "</label>";
          } else {
            var total = "<label class='label label-success'>" + amountStr + "</label>";
          }
          $("td:eq(0)", row).html(timestamp);
          $("td:eq(1)", row).html('<a href="/tx/' + txhash + '">' + txhash + '</a>');
          $("td:eq(2)", row).html(total);        
        },
      });
      setInterval( function () {
        rtable.api().ajax.reload(null, false);
      }, 45000 );
    });


ERROR…
/root/explorer/views/movement.jade:48 46| } 47| var rtable = $('#alerts-table').dataTable( { > 48| paginate: { 49| previous: "Anterior", 50| next: "Siguiente" 51| }, unexpected text {

**Comment by joeuhren** [comment]
2 problems with your code:

1. You missed adding the whole `language` object wrapper. All of your updated language changes must be nested inside an object called `language` like I showed in my pseudo-example above and as specified by the DataTables documentation
2. Pug/Jade uses a whitespace sensitive syntax, which means that it is extremely important that you line up everything exactly perfect or else you will get errors like what you are seeing

Try this:

```
  var rtable = $('#alerts-table').dataTable( { 
    language: {
      paginate: {
        previous: "Anterior",
        next: "Siguiente"
      }
    },
    autoWidth: false,      
    searching: false,
    ordering: false,
    //responsive: true,
    lengthChange: true,
    processing: true,
    serverSide: true,
    iDisplayLength: displayLengthMax,
    lengthMenu: lengthMenuOpts,
    //processing: true,
    ajax: '/ext/getlasttxsajax/#{min_amount}',
    rowCallback: function(row, data, index) {
      var blockindex = data[0]; //variables for better readability
      var blockhash = data[1]; //variables for better readability
      var txhash = data[2]; //variables for better readability
      var outputs = data[3]; //variables for better readability
      var amount = (data[4] / 100000000); //variables for better readability
      var amountParts = amount.toLocaleString('en',{'minimumFractionDigits':2,'maximumFractionDigits':8,'useGrouping':true}).split('.');
      var amountStr = amountParts[0] + '.<span class="decimal">' + amountParts[1] + '</span>';
      var timestamp = data[5]; //variables for better readability
      if (amount > '#{flagb}') {
        var total = "<label class='label label-danger'>" + amountStr + "</label>";
      } else if (amount > '#{flaga}') {
        var total = "<label class='label label-warning'>" + amountStr + "</label>";
      } else {
        var total = "<label class='label label-success'>" + amountStr + "</label>";
      }
      $("td:eq(0)", row).html(timestamp);
      $("td:eq(1)", row).html('<a href="/tx/' + txhash + '">' + txhash + '</a>');
      $("td:eq(2)", row).html(total);        
    },
  });
  setInterval( function () {
    rtable.api().ajax.reload(null, false);
  }, 45000 );
});
```

**Comment by dynarian** [comment]
Perfect!!!, thanks for U help.
Now i have the other translation...

**Comment by uaktags** [comment]
Closing, sounds complete now.
